src/authentication/models/User.ts

```diff
@@ -11,3 +11,10 @@ export interface IAuthUser extends User {
 export type IAuthProfile = NonNullable<Profile>;
 
 export const isOnlineUser = (user: User | IAuthUser): user is IAuthUser => user.hasOwnProperty('preferred_username');
+
+export interface IUserName {
+  id: number;
+  first_name: string;
+  last_name: string;
+  username: string;
+}
```

Comment:
Is this a subset of User or IAuthUser?
If so, perhaps a [Pick<T, K> type](https://www.typescriptlang.org/docs/handbook/utility-types.html#picktk) could be used?
`export const IUserName = Pick<IAuthUser, 'id' | 'first_name' | 'last_name' | 'username'>;`

Or is it a special type?

Reply:
This is a special type, made specifically to handle both users and anonymous users

Reply:
As I've understood them, IAuthUser is based on OW4's implementation of the OIDC-user model, thus extends it.
If this is the standard Django user model, I agree that it fits as it's own type! 🚀